### PR TITLE
Clarify bibliography page

### LINKED ISSUE #1581: Approach to bibliography -- request for clarity
### Summary

*The [Citing and Referencing](https://the-turing-way.netlify.app/community-handbook/style/style-citing.html) chapter is hard to follow and should be clarified for people who do not use GitHub or the tools employed.*

### What needs to be done?

- [ ] *Change pink text to links*
- [ ] *Explain what a Bibtex file is*
- [ ] *Provide tools for people to translate sites/references into the right formatting (such as [this one](https://doi2bib.org/) for DOIs)*

### Who can help?

* *Anybody who understands how citing works using these specific tools!*

---

### Updates

*Lorem ipsum dolor sit amet, consectetur adipiscing.*

### PR DESCRIPTION
**Pull request by erictleung**
### Summary

Fixes #1581 

### List of changes proposed in this PR (pull-request)

- Add direct link to bibliography for easy access
- Replace dash with em dash
- Indent bibliography directive to render correctly
- Differentiate similar named section
- Link to Sphinx BibTeX extension
- Add explanation for BibTeX files and share tools 

### What should a reviewer concentrate their feedback on?

- [ ] Everything looks ok?
- [ ] Anything else that needs to be clarified?

### Acknowledging contributors

- [x] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file:

### CODE REVIEW

**Comment by malvikasharan** [comment]
@erictleung This is great! Thank you for adding more details and links. I have made a minor correction and approved it. Let me know when you are ready to merge this. 🎊

**Review by malvikasharan — APPROVED** [review]

**Comment by erictleung** [comment]
@malvikasharan no problem! Thanks for your minor corrections. Looks good to me! :+1:
